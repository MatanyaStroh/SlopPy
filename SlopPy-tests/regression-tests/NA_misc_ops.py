# try out misc. ops on an NA object

x = 1 / 0
assert type(x) is NA

assert type(x.attr) is NA
x.attr2 = 5
assert type(x[42]) is NA
x[42] = 5
assert type(x[1:5]) is NA
x[1:5] = 5
assert len(x) == 1
assert type('foo' in x) is NA
assert type(x()) is NA
assert type(x(1,2,3)) is NA

