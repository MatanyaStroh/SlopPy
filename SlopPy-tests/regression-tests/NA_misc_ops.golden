{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/NA_misc_ops.py", line 3, in <module>
    x = 1 / 0

Locals: {'__builtins__': <module '__builtin__' (built-in)>, '__name__': '__main__', '__file__': 'regression-tests/NA_misc_ops.py', '__doc__': None, '__package__': None}
}
{NA_GetAttr | regression-tests/NA_misc_ops.py <module>:6}
{NA_subscript | regression-tests/NA_misc_ops.py <module>:8}
{NA_slice | regression-tests/NA_misc_ops.py <module>:10}
{NA_call | regression-tests/NA_misc_ops.py <module>:14}
{NA_call | regression-tests/NA_misc_ops.py <module>:15}
